Cookbook/Proposals/Base class for FeedbackLoops systems.md

```diff
@@ -0,0 +1,157 @@
+# Base class for FeedbackLoops systems
+
+* Author(s): Sergey Shulga
+* Review Manager: Anders Ha
+
+## Introduction
+
+Almost for all our screens, we implement ViewModels which are powered by the [ReactiveFeeedback](https://github.com/Babylonpartners/ReactiveFeedback).
+
+The current interface of VM looks like this
+
+```swift
+public protocol BoxViewModel: AnyObject {
+    associatedtype State
+    associatedtype Action
+
+    var state: Property<State> { get }
+
+    func send(action: Action)
+}
+```
+
+## Motivation
+
+One of the main issues with this approach
+
+1. We need to implement `input` feedback in pretty much every VM which may lead to errors if do not schedule input correctly, for this we have `Feedback.input()` convenience extension
+
+2. If the screen contains a lot of input field we end up having a lot of event's that are just mutate one property of the `State` e.g
+`Action.didChangeUserName(String)` `Action.didPickBirthDate(Date)`
+ 
+## Proposed solution
+1. Implementation of the feedback:
+
+```swift
+open class FeedbackLoop<State, Event, Action>: FeedbackLoopType {
+  public let state: Property<State>
+  private let input = Feedback<State, Update>.input()
+  private let actionTransformer: (Action) -> Event
+
+  public init(
+    initial: State,
+    reducer: @escaping (State, Event) -> State,
+    feedbacks: [Feedback<State, Event>],
+    scheduler: DateScheduler,
+    actionTransformer: @escaping (Action) -> Event
+  ) {
+    self.actionTransformer = actionTransformer
+    self.state = Property(
+      initial: initial,
+      scheduler: scheduler,
+      reduce: { (state: State, update: Update) -> State in
+        switch update {
+        case let .event(event):
+          return reducer(state, event)
+        case let .mutation(mutation):
+          return mutation.mutate(state)
+        }
+      },
+      feedbacks: feedbacks.map { $0.mapEvent(Update.event) }
+        .appending(self.input.feedback)
+    )
+  }
+
+  open func send(action: Action) {
+    self.input.observer(.event(actionTransformer(action)))
+  }
+
+  open func mutate<V>(keyPath: WritableKeyPath<State, V>, value: V) {
+    self.input.observer(.mutation(Mutation(keyPath: keyPath, value: value)))
+  }
+
+  private enum Update {
+    case event(Event)
+    case mutation(Mutation)
+  }
+
+  private struct Mutation {
```

Comment:
this will probably benefit a little bit from @dynamicCallable and this new proposal https://github.com/apple/swift-evolution/blob/master/proposals/0253-callable.md This way instead of `mutation.mutate(state)` we can just have `mutation(state)`.